File: app/lib/features/product/models/product.dart
/// 商品模型
class Product {
  final int id;
  final int categoryId;
  final String name;
  final String description;
  final double price;
  final int stock;
  final String coverImage;
  final List<String> images;
  final List<ProductSKU> skus;
  final List<Review> reviews;

  Product({
    required this.id,
    required this.categoryId,
    required this.name,
    required this.description,
    required this.price,
    required this.stock,
    required this.coverImage,
    required this.images,
    this.skus = const [],
    this.reviews = const [],
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['ID'], // GORM Model 使用 ID (大写)
      categoryId: json['category_id'] ?? 0,
      name: json['name'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      stock: json['stock'],
      coverImage: json['cover_image'],
      // 处理字符串数组
      images: List<String>.from(json['images'] ?? []),
      skus:
          (json['skus'] as List<dynamic>?)
              ?.map((e) => ProductSKU.fromJson(e))
              .toList() ??
          [],
      reviews:
          (json['reviews'] as List<dynamic>?)
              ?.map((e) => Review.fromJson(e))
              .toList() ??
          [],
    );
  }
}

class ProductSKU {
  final int id;
  final String name;
  final String specs;
  final double price;
  final int stock;

  ProductSKU({
    required this.id,
    required this.name,
    required this.specs,
    required this.price,
    required this.stock,
  });

  factory ProductSKU.fromJson(Map<String, dynamic> json) {
    return ProductSKU(
      id: json['ID'],
      name: json['name'],
      specs: json['specs'],
      price: (json['price'] as num).toDouble(),
      stock: json['stock'],
    );
  }
}

class Review {
  final int id;
  final int userId;
  final String userName;
  final String userAvatar;
  final String content;
  final int rating;
  final String createdAt;

  Review({
    required this.id,
    required this.userId,
    required this.userName,
    required this.userAvatar,
    required this.content,
    required this.rating,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    final user = json['user'] ?? {};
    return Review(
      id: json['ID'],
      userId: json['user_id'],
      userName: user['username'] ?? '匿名用户',
      userAvatar: user['avatar'] ?? '',
      content: json['content'],
      rating: json['rating'],
      createdAt: json['CreatedAt'] ?? '',
    );
  }
}

File: backend/models/product.go
package models

import (
	"github.com/lib/pq"
	"gorm.io/gorm"
)

// Category 表示商品分类
type Category struct {
	gorm.Model
	Name      string `json:"name"`       // 分类名称
	SortOrder int    `json:"sort_order"` // 排序权重
	Icon      string `json:"icon"`       // 分类图标
}

// Product 表示商品信息
// 包含商品的基本属性和关联的 SKU
type Product struct {
	gorm.Model
	Name        string         `json:"name"`                                          // 商品名称
	Description string         `json:"description"`                                   // 商品描述
	Price       float64        `json:"price"`                                         // 商品基础价格
	Stock       int            `json:"stock"`                                         // 总库存
	CoverImage  string         `json:"cover_image"`                                   // 封面图片 URL
	Images      pq.StringArray `gorm:"type:text[]" json:"images"`                     // 商品轮播图列表 (PostgreSQL 数组类型)
	CategoryID  uint           `json:"category_id"`                                   // 分类 ID
	Status      int            `gorm:"default:1" json:"status"`                       // 商品状态: 1-上架, 0-下架
	SKUs        []ProductSKU   `gorm:"foreignKey:ProductID" json:"skus"`              // 关联的 SKU 列表
	Reviews     []Review       `gorm:"foreignKey:ProductID" json:"reviews,omitempty"` // 关联的评价列表
}

// ProductSKU 表示商品的库存量单位 (Stock Keeping Unit)
// 用于管理商品的不同规格 (如颜色、尺寸)
type ProductSKU struct {
	gorm.Model
	ProductID uint    `json:"product_id"` // 关联的商品 ID
	Name      string  `json:"name"`       // SKU 名称 (如 "红色 XL")
	Specs     string  `json:"specs"`      // 规格详情 JSON 字符串
	Price     float64 `json:"price"`      // SKU 价格
	Stock     int     `json:"stock"`      // SKU 库存
	Image     string  `json:"image"`      // SKU 图片
}

// CartItem 表示购物车中的一项
type CartItem struct {
	gorm.Model
	UserID    uint    `json:"user_id"`                               // 关联的用户 ID
	ProductID uint    `json:"product_id"`                            // 关联的商品 ID
	Product   Product `json:"product"`                               // 预加载的商品信息
	SKUID     uint    `gorm:"column:sku_id;default:0" json:"sku_id"` // 关联的 SKU ID (如果商品有规格)
	Quantity  int     `json:"quantity"`                              // 购买数量
	Selected  bool    `gorm:"default:true" json:"selected"`          // 是否选中
}
